functions/sql_queries.R

```diff
@@ -0,0 +1,282 @@
+
+       "and ({{fn left(cause_of_death_code_3, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_3, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+
+       "and ({{fn left(cause_of_death_code_4, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_4, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+
+       "and ({{fn left(cause_of_death_code_5, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_5, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+
+       "and ({{fn left(cause_of_death_code_6, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_6, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+
+       "and ({{fn left(cause_of_death_code_7, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_7, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+
+       "and ({{fn left(cause_of_death_code_8, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_8, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+
+       "and ({{fn left(cause_of_death_code_9, 3)}} is null or ",
+       "{{fn left(cause_of_death_code_9, 3)}} not in ",
+       "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')})) ",
+       
+       # Select deaths in reporting period
+       "and (date_of_death between ",
+       "to_date({shQuote(extract_start, type = 'sh')}, 'yyyy-mm-dd') ",
+       "and to_date({shQuote(extract_end, type = 'sh')}, 'yyyy-mm-dd')) ",
+       
+       # Exclude deaths with missing postcode
+       "and postcode is not null"
+       
+  )
+  
+}
+  
+  
+### 2 - SMR01 query ----
+
+smr01_query <- function(extract_start, 
+                        extract_end,
+                        external_causes,
+                        gls){
+  
+  data <- if_else(gls == TRUE, "smr01_1e_pi", "smr01_pi")
+  
+  extract_start_smr <- extract_start - months(6)
+  
+  glue(
+    "select s.link_no, s.gls_cis_marker, ",
+    "s.admission_date, s.discharge_date, d.date_of_death ",
+    
+    "from analysis.{data} s, analysis.gro_deaths_c d ",
+    
+    # Only extract SMR records with matching death record
+    "where s.link_no = d.link_no ",
+    
+    # Inpatients only
+    "and s.inpatient_daycase_identifier = 'I' ",
+    
+    # Select records in reporting period (and six months before)
+    "and s.discharge_date between ",
+    "to_date({shQuote(extract_start_smr, type = 'sh')}, 'yyyy-mm-dd') ",
+    "and to_date({shQuote(extract_end, type = 'sh')}, 'yyyy-mm-dd') ",
+    
+    # Exclude external causes of death
+    "and {{fn left(d.underlying_cause_of_death, 3)}} not in ",
+    "({paste0(shQuote(external_causes, type = 'sh'), collapse = ',')}) ",
+
+    "and ({{fn left(cause_of_death_code_0, 3)}} is null or ",
```

Comment:
Just want to double check do we not need "d." in front of cause_of_death_code_0?